practice-app/backend/disasterresponse/src/main/java/com/a1/disasterresponse/service/GeocodeService.java

```diff
@@ -20,20 +22,19 @@ public GeocodeService(@Value("${google.api.key}") String apiKey) {
                 .build();
     }
 
-    public LatLng geocodeAddress(String address) throws Exception {
+    public GeocodeData geocodeAddress(String address) throws Exception {
         GeocodingResult[] results = GeocodingApi.geocode(geoApiContext, address).await();
         if (results.length > 0) {
-            return results[0].geometry.location;
+            return new GeocodeData(results[0].geometry.location.lat, results[0].geometry.location.lng, address);
         }
         return null;
     }
 
-    public String reverseGeocodeCoordinates(double lat, double lng) throws Exception {
-        LatLng latLng = new LatLng(lat, lng);
-        GeocodingResult[] results = GeocodingApi.reverseGeocode(geoApiContext, latLng).await();
+    public GeocodeData reverseGeocodeCoordinates(double lat, double lng) throws Exception {
+        GeocodingResult[] results = GeocodingApi.reverseGeocode(geoApiContext, lat, lng).await();
```

Comment:
Harun, The method reverseGeocode(GeoApiContext, LatLng) in the type GeocodingApi is not applicable for the arguments (GeoApiContext, double, double). @alitpc25 Please check it out before merge. :)

Reply:
Resolved 👍